csrc/scheduler/registry.cpp

```diff
@@ -81,14 +81,16 @@ SchedulerRuntimeInfo::SchedulerRuntimeInfo(
       // find and push discontiguous stride
       int64_t dtype_size = dataTypeSize(input_tv->dtype());
       input_discontig_strides_[fusion_inp] = {};
-      int64_t dims = (int64_t)alloc_strides.size();
+      auto dims = static_cast<int64_t>(alloc_strides.size());
       int64_t expected_stride = 1;
       for (int64_t dim = dims - 1; dim >= 0; dim--) {
         auto size = alloc_sizes.at(dim);
-        if (size <= 1) {
+        auto stride = alloc_strides.at(dim);
+        // Skip broadcast dimensions because they don't affect contiguity.
+        if (size <= 1 || stride == 0) {
```

Comment:
@jjsjann123 would you mind clarifying https://github.com/NVIDIA/Fuser/pull/2815/commits/3ff26bccab7fdcfb333a5043f8e16d19d98284d6#r1731934823? I don't follow the difference between isBroadcast and checking the size and the stride. I expect enough validations elsewhere to ensure a broadcast dimension must be size-1 and/or stride-0.

Reply:
> I expect enough validations elsewhere

That's a fair ask. Looks like we are safe from the python side. https://github.com/NVIDIA/Fuser/blob/742f24c9c00c8a25d631751e7c8a05a2fa675e55/csrc/python_frontend/python_bindings.cpp#L233-L235

earlier I was concerned about the c++ API. I think user could present a set of inputs with stride == 0 without marking the dimension as broadcast. Semantically, they are not promising a dimension is broadcast, though the execution input could be interpreted as such.
So feels like the broadcast definition is a more robust check, instead of checking the explicit size/stride, which is only retrieved from the runtime inputs.

Reply:
I don't have a super strong stance on this one from the python use case. I'm happy with putting a NOTE in the comment and proceed forward with this as-is.

This does feel a bit scary on the c++ side though....

Reply:
BTW, not saying it's your modification that introduced the problem, the old code is using `size <= 1` here, so the crime has already been committed. :laughing: